lib/checks/checks.go

```diff
@@ -319,119 +329,145 @@ func (r *checker) Run(ctx context.Context) error {
 		return nil
 	}
 
-	var errors []error
+	var failed []*agentpb.Probe
 
 	// check each server against its profile
 	for _, server := range r.Servers {
-		errors = append(errors, r.CheckNode(ctx, server))
+		failed = append(failed, r.CheckNode(ctx, server)...)
 	}
 
 	// run checks that take all servers into account
-	errors = append(errors, r.CheckNodes(ctx, r.Servers))
+	failed = append(failed, r.CheckNodes(ctx, r.Servers)...)
+
+	if len(failed) != 0 {
+		return trace.BadParameter("The following checks failed:\n%v",
+			FormatFailedChecks(failed))
+	}
 
-	return trace.NewAggregate(errors...)
+	return nil
 }
 
 // CheckNode executes checks for the provided individual server.
-func (r *checker) CheckNode(ctx context.Context, server Server) error {
+func (r *checker) CheckNode(ctx context.Context, server Server) (failed []*agentpb.Probe) {
```

Comment:
Changed this (and `CheckNodes` below) from returning a generic error to a list of failed probes instead so various callers have a chance to format the errors properly.